[CI] OldMappingsIT testSearchFieldsOnPlaceholderField failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/57qtf7byqdzb6/tests/:x-pack:qa:repository-old-versions:javaRestTestAfterRestart%235_0_0/org.elasticsearch.oldrepos.OldMappingsIT/testSearchFieldsOnPlaceholderField
**Reproduction line:**
```
./gradlew ':x-pack:qa:repository-old-versions:javaRestTestAfterRestart#5_0_0' --tests "org.elasticsearch.oldrepos.OldMappingsIT.testSearchFieldsOnPlaceholderField" -Dtests.seed=2458E0756602C685 -Dtests.locale=pl -Dtests.timezone=America/Curacao -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.oldrepos.OldMappingsIT&tests.test=testSearchFieldsOnPlaceholderField


**Failure excerpt:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:42911], URI [/custom/_search], status line [HTTP/1.1 404 Not Found]
{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [custom]","resource.type":"index_or_alias","resource.id":"custom","index_uuid":"_na_","index":"custom"}],"type":"index_not_found_exception","reason":"no such index [custom]","resource.type":"index_or_alias","resource.id":"custom","index_uuid":"_na_","index":"custom"},"status":404}

  at __randomizedtesting.SeedInfo.seed([2458E0756602C685:968AAE4EFC2536E4]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:347)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)
  at org.elasticsearch.oldrepos.OldMappingsIT.testSearchFieldsOnPlaceholderField(OldMappingsIT.java:310)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

The first failure was a periodic build with this commit: https://github.com/elastic/elasticsearch/commit/0cf140319f1d1b5dc8f18565fd2874ba462b8e34

But that commit had nothing to do with any of these files. 

From what I can tell, the index itself just isn't created at all for the tests.

The logs: 
[repository-old-versions.zip](https://github.com/elastic/elasticsearch/files/13322442/repository-old-versions.zip)
interesting trace
```
[2023-11-10T09:16:15,186][WARN ][o.e.c.NodeConnectionsService] [5-0-0-1] failed to connect to {5-0-0-0}{W2lP1MB9Srq50SwHPHh1Aw}{jtFjVMuVSkK4204cPuGY8w}{5-0-0-0}{127.0.0.1}{127.0.0.1:45451}{cdfhilmrstw}{8.12.0}{7000099-8500003}{ml.allocated_processors=32, transform.config_version=10.0.0, xpack.installed=true, ml.config_version=11.0.0, testattr=test, ml.allocated_processors_double=32.0, upgraded=true, ml.max_jvm_size=536870912, ml.machine_memory=126604357632} (tried [1] times) org.elasticsearch.transport.ConnectTransportException: [5-0-0-0][127.0.0.1:45451] connect_exception
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1156)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener$FailureResult.complete(SubscribableListener.java:322)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:234)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:133)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:108)
	at org.elasticsearch.transport.netty4@8.12.0-SNAPSHOT/org.elasticsearch.transport.netty4.Netty4TcpChannel.addConnectListener(Netty4TcpChannel.java:140)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TcpTransport.initiateConnection(TcpTransport.java:417)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TcpTransport.lambda$openConnection$2(TcpTransport.java:378)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:368)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TcpTransport.openConnection(TcpTransport.java:365)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.ClusterConnectionManager.internalOpenConnection(ClusterConnectionManager.java:364)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.ClusterConnectionManager.connectToNodeOrRetry(ClusterConnectionManager.java:212)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.ClusterConnectionManager.connectToNode(ClusterConnectionManager.java:111)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:490)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:470)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.NodeConnectionsService$ConnectionTarget.doConnect(NodeConnectionsService.java:288)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.NodeConnectionsService$ConnectionTarget.lambda$connect$0(NodeConnectionsService.java:235)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.NodeConnectionsService.connectToNodes(NodeConnectionsService.java:124)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.connectToNodesAsync(ClusterApplierService.java:521)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.connectToNodesAndWait(ClusterApplierService.java:511)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:485)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:429)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:154)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:916)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:217)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:183)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.elasticsearch.common.util.concurrent.UncategorizedExecutionException: Failed execution
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.wrapAsExecutionException(SubscribableListener.java:215)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ListenableFuture.wrapException(ListenableFuture.java:38)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ListenableFuture.wrapException(ListenableFuture.java:27)
	at org.elasticsearch.server@8.12.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.onFailure(SubscribableListener.java:178)
	at org.elasticsearch.transport.netty4@8.12.0-SNAPSHOT/org.elasticsearch.transport.netty4.Netty4TcpChannel.lambda$addListener$0(Netty4TcpChannel.java:62)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.94.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	... 1 more
Caused by: java.util.concurrent.ExecutionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.0.1/127.0.0.1:45451
	... 21 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.0.1/127.0.0.1:45451
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
	at io.netty.transport@4.1.94.Final/io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.94.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1583)

[2023-11-10T09:16:15.287088188Z] [BUILD] Stopping node
```

--- a/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/LegacyRestTestBasePlugin.java
+++ b/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/LegacyRestTestBasePlugin.java
@@ -105,7 +105,6 @@ public class LegacyRestTestBasePlugin implements Plugin<Project> {
                     );
                 }
             }
-            configureCacheability(restIntegTestTask);
         });
 
         project.getTasks()
@@ -127,16 +126,17 @@ public class LegacyRestTestBasePlugin implements Plugin<Project> {
                     t.getClusters().forEach(c -> c.plugin(bundle));
                 }
             });
+            configureCacheability(t);
         });
     }
 
-    private void configureCacheability(RestIntegTestTask restIntegTestTask) {
+    private void configureCacheability(StandaloneRestIntegTestTask testTask) {
         TaskContainer tasks = project.getTasks();
         Spec<Task> taskSpec = t -> tasks.withType(StandaloneRestIntegTestTask.class)
             .stream()
-            .filter(task -> task != restIntegTestTask)
-            .anyMatch(task -> Collections.disjoint(task.getClusters(), restIntegTestTask.getClusters()) == false);
-        restIntegTestTask.getOutputs()
+            .filter(task -> task != testTask)
+            .anyMatch(task -> Collections.disjoint(task.getClusters(), testTask.getClusters()) == false);
+        testTask.getOutputs()
             .doNotCacheIf(
                 "Caching disabled for this task since it uses a cluster shared by other tasks",
                 /*
@@ -147,7 +147,7 @@ public class LegacyRestTestBasePlugin implements Plugin<Project> {
                  */
                 taskSpec
             );
-        restIntegTestTask.getOutputs().upToDateWhen(new NotSpec(taskSpec));
+        testTask.getOutputs().upToDateWhen(new NotSpec(taskSpec));
     }
 
     private String systemProperty(String propName) {

